sdk-cpp/include/do_exceptions.h

```diff
@@ -1,21 +1,45 @@
 #ifndef _DELIVERY_OPTIMIZATION_DO_EXCEPTIONS_H
 #define _DELIVERY_OPTIMIZATION_DO_EXCEPTIONS_H
 
-#include <exception>
 #include <stdint.h>
 #include <system_error>
 
-#if defined(DO_INTERFACE_COM)
-#include <winerror.h>   // FAILED macro
+#if (!DO_DISABLE_EXCEPTIONS)
+#include <exception>
 #endif
 
 namespace microsoft
 {
 namespace deliveryoptimization
 {
 
+//These error macros are defined behind guards so that they do not redefine the macro for for platforms/toolchains in which they already exist
+#ifndef RETURN_RES_IF_FAILED
+#define RETURN_RES_IF_FAILED(res)  {  \
+    int32_t __res = (res);            \
+    if(FAILED(__res)) return __res;   \
+}
+#endif
+
+#ifndef FACILITY_DELIVERY_OPTIMIZATION 
```

Comment:
>FACILITY_DELIVERY_OPTIMIZATION [](http://example.com/codeflow?start=8&length=30)

Can you pull out these definitions from client-lite/src/include/hresult_helpers.h into a new header in common/inc/basic_error_helpers.h?

Reply:
Sounds good, that file is pulled from WIl right? So we're good to use it over